maxminddb/extension/maxminddb.c

```diff
@@ -209,6 +211,28 @@ static PyObject *Reader_close(PyObject *self, PyObject *UNUSED(args))
         mmdb_obj->mmdb = NULL;
     }
 
+    mmdb_obj->closed = Py_True;
+
+    Py_RETURN_NONE;
+}
+
+static PyObject *Reader__enter__(PyObject *self, PyObject *UNUSED(args))
+{
+    Reader_obj *mmdb_obj = (Reader_obj *)self;
+
+    if(mmdb_obj->closed == Py_True) {
+        PyErr_SetString(PyExc_ValueError,
+                        "Attempt to reopen a closed MaxMind DB.");
+        return NULL;
+    }
+
+    Py_INCREF(self);
```

Comment:
If you are incrementing the reference count here, I think you would need to decrement it in `Reader__exit__`.

Was this incrementing necessary to prevent errors? Without checking, I am not sure whether the Python interpreter increments the reference count for us or whether this is necessary.